dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/privacy/PrivacyConsentProvider.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.privacy
+
+import java.util.LinkedList
+
+internal class PrivacyConsentProvider(consent: Consent = Consent.PENDING) :
+    ConsentProvider {
+
+    private val callbacks: LinkedList<ConsentProviderCallback> = LinkedList()
+
+    @Volatile
+    private var lastConsent: Consent
+
+    // region ConsentProvider
+
+    init {
+        lastConsent = consent
+    }
+
+    override fun getLastConsent(): Consent {
+        return lastConsent
+    }
+
+    // We need to synchronize everything as we are not sure from which thread the client
+    // will update the consent and initialize the SDK.
+    @Synchronized
+    override fun setConsent(consent: Consent) {
+        notifyCallbacks(consent)
+        lastConsent = consent
```

Comment:
IMO we should first update state, then notify callbacks. Otherwise, the state obtained from callback implementation will be inconsistent.